src/test-komodo/test_block.cpp

```diff
@@ -67,147 +69,96 @@ TEST(block_tests, TestConnectWithoutChecks)
-    aliceToBobMutable.vin[0].scriptSig << alice->Sign(hash, SIGHASH_ALL);
-    CTransaction aliceToBobTx(aliceToBobMutable);
+    {
+        CCoinControl useThisTransaction;
+        COutPoint tx(fundAlice.transaction.GetHash(), 1);
+        useThisTransaction.Select(tx);
+        TransactionInProcess aliceToBob = alice->CreateSpendTransaction(bob, 10000, 5000, useThisTransaction);
+        EXPECT_TRUE(alice->CommitTransaction(aliceToBob.transaction, aliceToBob.reserveKey));
+    }
     // alice attempts to double spend the vout and send something to charlie
-    CMutableTransaction aliceToCharlieMutable;
-    CTxIn aliceIn2;
-    aliceIn2.prevout.hash = fundAlice.GetHash();
-    aliceIn2.prevout.n = 0;
-    aliceToCharlieMutable.vin.push_back(aliceIn2);
-    CTxOut charlieOut;
-    charlieOut.scriptPubKey = GetScriptForDestination(charlie->GetPubKey());
-    charlieOut.nValue = 10000;
-    aliceToCharlieMutable.vout.push_back(charlieOut);
-    CTxOut aliceRemainder2;
-    aliceRemainder2.scriptPubKey = GetScriptForDestination(alice->GetPubKey());
-    aliceRemainder2.nValue = fundAlice.vout[0].nValue - 10000;
-    aliceToCharlieMutable.vout.push_back(aliceRemainder2);
-    hash = SignatureHash(fundAlice.vout[0].scriptPubKey, aliceToCharlieMutable, 0, SIGHASH_ALL, 0, 0);
-    aliceToCharlieMutable.vin[0].scriptSig << alice->Sign(hash, SIGHASH_ALL);
-    CTransaction aliceToCharlieTx(aliceToCharlieMutable);
-    // construct the block
-    CBlock block;
-    // first a coinbase tx
-    auto consensusParams = Params().GetConsensus();
-    CMutableTransaction txNew = CreateNewContextualCMutableTransaction(consensusParams, newHeight);
-    txNew.vin.resize(1);
-    txNew.vin[0].prevout.SetNull();
-    txNew.vin[0].scriptSig = (CScript() << newHeight << CScriptNum(1)) + COINBASE_FLAGS;
-    txNew.vout.resize(1);
-    txNew.vout[0].nValue = GetBlockSubsidy(newHeight,consensusParams);
-    txNew.nExpiryHeight = 0;
-    block.vtx.push_back(CTransaction(txNew));
-    // then the actual txs
-    block.vtx.push_back(fundAlice);
-    block.vtx.push_back(aliceToBobTx);
-    block.vtx.push_back(aliceToCharlieTx);
-    CValidationState state;
-    // create a new CBlockIndex to forward to ConnectBlock
-    auto index = chain.GetIndex();
-    CBlockIndex newIndex;
-    newIndex.pprev = index;
-    EXPECT_FALSE( chain.ConnectBlock(block, state, &newIndex, true, false) );
-    EXPECT_EQ(state.GetRejectReason(), "bad-txns-inputs-missingorspent");
+    {
+        CCoinControl useThisTransaction;
+        COutPoint tx(fundAlice.transaction.GetHash(), 1);
+        useThisTransaction.Select(tx);
+        TransactionInProcess aliceToCharlie = alice->CreateSpendTransaction(charlie, 10000, 5000, useThisTransaction);
+        CValidationState state;
+        EXPECT_FALSE(alice->CommitTransaction(aliceToCharlie.transaction, aliceToCharlie.reserveKey, state));
+        EXPECT_EQ(state.GetRejectReason(), "mempool conflict");
```

Comment:
Need to investigate: Why did this change? Should the mempool accept transactions that spend the same prevout? It seems that it used to. Could this be related to the COINBASE_MATURITY change? Doubtful, but possible.

Reply:
The old code skipped the mempool validation. The new code is more accurate.